src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryFormatterWriter.cs

```diff
@@ -79,7 +80,18 @@ internal void WriteEnd()
 
         internal void WriteTimeSpan(TimeSpan value) => WriteInt64(value.Ticks);
 
-        internal void WriteDateTime(DateTime value) => WriteInt64(value.Ticks); // in desktop, this uses ToBinaryRaw
+        internal void WriteDateTime(DateTime value)
+        {
+            // In .NET Framework, BinaryFormatter is able to access DateTime's ToBinaryRaw,
+            // which just returns the value of its sole Int64 dateData field.  Here, we don't
+            // have access to that member (which doesn't even exist anymore, since it was only for
+            // BinaryFormatter, which is now in a separate assembly).  To address that,
+            // we access the sole field directly via an unsafe cast; ideally this would
+            // just use Unsafe.As, but since we can't use that here due to not having a netcoreapp
+            // build of System.Runtime.CompilerServices.Unsafe.dll, we instead go through span.
```

Comment:
https://github.com/dotnet/corefx/blob/master/src/System.Runtime.CompilerServices.Unsafe/src/Configurations.props#L11 says that we have netcoreapp build

Reply:
Hmm, you're right, not sure what I did incorrectly before, but I was able to use it successfully now.

Reply:
Actually, I think some packaging verification may get unhappy about this.

System.Runtime.CompilerServices.Unsafe is not include inbox for NetCoreApp right now. I do not see a problem with going that. It can be on the same plan as System.Collections.Immutable or System.Reflection.Metadata.

Reply:
> Actually, I think some packaging verification may get unhappy about this.

Ah, that must have been what I hit previously.

I'm going to revert back to using MemoryMarshal for now and open an issue about Unsafe.

Reply:
Yeah:
```
/Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/pkg/frameworkPackage.targets(124,5): error : Assembly 'System.Runtime.Serialization.Formatters' is missing dependency 'System.Runtime.CompilerServices.Unsafe' [/Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
```